### PR TITLE
Add argument validation for Slice.indexOfByte()

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: src/main/java/io/airlift/slice/Slice.java [hunk 1]
```diff
@@ -888,7 +888,7 @@ public Slice slice(int index, int length)
 
     public int indexOfByte(int b)
     {
-        b &= 0xFF;
+        checkArgument((b >> Byte.SIZE) == 0, "byte value out of range");
```
Comment (findepi) [inline src/main/java/io/airlift/slice/Slice.java]: if the advantage of this compare to `0 <= b && b < 256`?

Comment (findepi) [reply src/main/java/io/airlift/slice/Slice.java]: we should provide an overload that takes `byte`

Comment (electrum) [reply src/main/java/io/airlift/slice/Slice.java]: There’s no branching, but it’s hard to say if it’s actually faster. This is how it’s done in Guava.

Comment (electrum) [reply src/main/java/io/airlift/slice/Slice.java]: Good idea, added an overload

**Pull request merged**